### ISSUE TITLE
Refresh of unit matches first request, never subsequent requests.

### ISSUE THREAD

**Issue by thomaslsimpson** [prebid/Prebid.js#831]
## Type of issue
Bug

## Description

When a refresh is run on a single unit, the result references the first request every time, regardless of subsequent request. This results in some of the information in the response being incorrect, including the requestId and timer information, which ends up coming from the first request, not the subsequent request.

If you use the sample on your web page for doing refresh and then look at the events returned, you will see that they do not match. (I copied it all below.)

```
// from line 85 of bidmanager.js
function getBidderRequest(bidder, adUnitCode) {                                                            
  return $$PREBID_GLOBAL$$._bidsRequested.find(request => {                                                
    return request.bids                                                                                    
        .filter(bid => bid.bidder === bidder && bid.placementCode === adUnitCode).length > 0;              
  }) || { start: null, requestId: null };                                           
}                                                                                   
```
The only two parameters are bidder and adUnitCode, so the second time around, we get the first one, not the second.

## Steps to reproduce

```
<html>
<head>
  <!-- Prebid Config Section START -->
  <!-- Make sure this is inserted before your GPT tag -->
  <script> 
    var PREBID_TIMEOUT = 700;

    var adUnits = [{
        code: 'div-gpt-ad-1460505748561-0',
        sizes: [[300, 250], [300,600]],
        bids: [{
            bidder: 'appnexus',
            params: {
               placementId: '4799418'
            }
        }]
    },{
        code: 'div-gpt-ad-1460505661639-0',
        sizes: [[728, 90], [970, 90]],
        bids: [{
            bidder: 'appnexus',
            params: {
               placementId: '4799418'
            }
        }]
    }];
    
    var pbjs = pbjs || {};
    pbjs.que = pbjs.que || [];

    pbjs.que.push(function(){
      pbjs.onEvent("bidRequested", function (e) {                                          
        console.log("PBJSEVENT: BID_REQ", e.requestId);
      });                                                                                         

      pbjs.onEvent("bidResponse", function (e) {                                           
        console.log("PBJSEVENT: BID_RES", e.requestId);                      
      });                                                                                         
    })

    var slot300;
    var slot728;
    
    function refreshSlot(slot) {
        pbjs.que.push(function() {
         pbjs.requestBids({
           timeout: PREBID_TIMEOUT,
           adUnitCodes: [slot.getSlotElementId()],
           bidsBackHandler: function() {
             pbjs.setTargetingForGPTAsync([slot.getSlotElementId()]);
             googletag.pubads().refresh([slot]); 
           }
         });
        }); 
    }

    function refreshSlot1() {
        refreshSlot(slot300);
    }

    function refreshSlot2() {
        refreshSlot(slot728);
    }
  </script>
  <!-- Prebid Config Section END -->
  
  <!-- Prebid Boilerplate Section START. No Need to Edit. -->
  <script type="text/javascript" src="//acdn.adnxs.com/prebid/not-for-prod/prebid.js" async></script>
  <script>
    var googletag = googletag || {};
    googletag.cmd = googletag.cmd || [];
    googletag.cmd.push(function() {
        googletag.pubads().disableInitialLoad();
    });

    pbjs.que.push(function() {
        pbjs.addAdUnits(adUnits);
        pbjs.requestBids({
            bidsBackHandler: sendAdserverRequest
        });
    });

    function sendAdserverRequest() {
        if (pbjs.adserverRequestSent) return;
        pbjs.adserverRequestSent = true;
        googletag.cmd.push(function() {
            pbjs.que.push(function() {
                pbjs.setTargetingForGPTAsync();
                googletag.pubads().refresh();
            });
        });
    }

    setTimeout(function() {
        sendAdserverRequest();
    }, PREBID_TIMEOUT);
  </script>
  <!-- Prebid Boilerplate Section END -->

  <script>
    (function () {
        var gads = document.createElement('script');
        gads.async = true;
        gads.type = 'text/javascript';
        var useSSL = 'https:' == document.location.protocol;
        gads.src = (useSSL ? 'https:' : 'http:') +
                '//www.googletagservices.com/tag/js/gpt.js';
        var node = document.getElementsByTagName('script')[0];
        node.parentNode.insertBefore(gads, node);
    })();
  </script>

  <script>
    googletag.cmd.push(function () {
        slot300 = googletag.defineSlot('/19968336/header-bid-tag-0', [[300, 250], [300, 600]], 'div-gpt-ad-1460505748561-0').addService(googletag.pubads());

        slot728 = googletag.defineSlot('/19968336/header-bid-tag1', [[728, 90], [970, 90]], 'div-gpt-ad-1460505661639-0').addService(googletag.pubads());

        googletag.pubads().enableSingleRequest();
        googletag.enableServices();
    });
  </script>
</head>

<body>
<h2>Prebid.js Test</h2>
<h5>Div-1</h5>
<button onclick="refreshSlot1()">Refresh 300x250 Ad Unit</button>
<div id='div-gpt-ad-1460505748561-0'>
    <script type='text/javascript'>
        googletag.cmd.push(function() { googletag.display('div-gpt-ad-1460505748561-0'); });
    </script>
</div>

<h5>Div-2</h5>
<button onclick="refreshSlot2()">Refresh 728x90 Ad Unit</button>
<div id='div-gpt-ad-1460505661639-0'>
    <script type='text/javascript'>
        googletag.cmd.push(function() { googletag.display('div-gpt-ad-1460505661639-0'); });
    </script>
</div>

</body>
</html>

```

Load the page. Click One of the refresh buttons. Look at the console window.

### Expected results
(In console window, we see a single request and two responses, with matching request ID:)
```
PBJSEVENT: BID_REQ f9cec624-180d-4037-9184-d44bc87f28f5
PBJSEVENT: BID_REQ f9cec624-180d-4037-9184-d44bc87f28f5
PBJSEVENT: BID_RES f9cec624-180d-4037-9184-d44bc87f28f5
```
After we click the reload, we get (in console):
```
PBJSEVENT: BID_REQ 1e3f2cbb-346e-4afb-9240-ca2755cc729d
PBJSEVENT: BID_RES 1e3f2cbb-346e-4afb-9240-ca2755cc729d
```

### Actual results
(In console window, we see a single request and two responses, with matching request ID:)
```
PBJSEVENT: BID_REQ f9cec624-180d-4037-9184-d44bc87f28f5
PBJSEVENT: BID_REQ f9cec624-180d-4037-9184-d44bc87f28f5
PBJSEVENT: BID_RES f9cec624-180d-4037-9184-d44bc87f28f5
```
After we click the reload, we get (in console):
```
PBJSEVENT: BID_REQ 1e3f2cbb-346e-4afb-9240-ca2755cc729d
PBJSEVENT: BID_RES f9cec624-180d-4037-9184-d44bc87f28f5
```
Note the last line with the request id: **f9cec624-180d-4037-9184-d44bc87f28f5**
It should be: **1e3f2cbb-346e-4afb-9240-ca2755cc729d**

## Platform details

Current 0.15.3-pre (but it is in older versions.) 
Macbook (not relevant here)
Chrome (also not relevant)

## Other information

It seems to me the issue is:
```
// from line 85 of bidmanager.js
function getBidderRequest(bidder, adUnitCode) {
```
The list of bid requests is stored in an array, when only the first is being returned. So, of course, after a subsequent request is put in, the first request is still in the list and matches `(bidder,adUnitCode)` just as well as the subsequent calls.

In case I'm wrong about the problem, you may also note that the problem only happens when refreshing a unit that has already has a request. If you refresh everything, it does not happen. If you use new units in subsequent calls, it does not happen.

**Comment by protonate** [comment]
Thanks for the report. This is the same issue mentioned in #825 and should be resolved with #828. Please test on that branch to confirm it resolves your issue.

**Comment by thomaslsimpson** [comment]
Thanks - don't know how that didn't turn up in a search? Looks like the same exact thing. I'll be sure to test.
